src/plugins/gimbal/gimbal_impl.cpp

```diff
@@ -24,162 +24,229 @@ GimbalImpl::~GimbalImpl()
+                } else {
+                    _parent->call_user_callback([callback, result]() {
+                        callback(gimbal_result_from_command_result(result));
+                    });
+                }
+            },
+            0); // any component
+    }
+} // namespace mavsdk
 
-void GimbalImpl::process_gimbal_manager_information(const mavlink_message_t& message)
+void GimbalImpl::disable()
 {
-    mavlink_gimbal_manager_information_t gimbal_manager_information;
-    mavlink_msg_gimbal_manager_information_decode(&message, &gimbal_manager_information);
-
-    if (_protocol_cookie != nullptr) {
-        LogDebug() << "Using Gimbal Version 2 as gimbal manager information for gimbal device "
-                   << static_cast<int>(gimbal_manager_information.gimbal_device_id)
-                   << " was discovered";
-
-        _parent->unregister_timeout_handler(_protocol_cookie);
-        _protocol_cookie = nullptr;
-        _gimbal_protocol.reset(new GimbalProtocolV2(
-            *_parent, gimbal_manager_information, message.sysid, message.compid));
-    }
+    _gimbal_protocol.reset(nullptr);
 }
 
 Gimbal::Result GimbalImpl::set_pitch_and_yaw(float pitch_deg, float yaw_deg)
 {
-    wait_for_protocol();
+    if (_gimbal_protocol == nullptr) {
+        LogErr() << "Gimbal plugin not prepared! Call `prepare()` first!";
+        return Gimbal::Result::Error;
+    }
 
     return _gimbal_protocol->set_pitch_and_yaw(pitch_deg, yaw_deg);
 }
 
 void GimbalImpl::set_pitch_and_yaw_async(
     float pitch_deg, float yaw_deg, Gimbal::ResultCallback callback)
 {
-    wait_for_protocol_async(
-        [=]() { _gimbal_protocol->set_pitch_and_yaw_async(pitch_deg, yaw_deg, callback); });
+    if (_gimbal_protocol == nullptr) {
+        _parent->call_user_callback([callback]() {
+            LogErr() << "Gimbal plugin not prepared! Call `prepare()` first!";
+            callback(Gimbal::Result::Error);
+        });
+
+        return;
+    }
+
+    _gimbal_protocol->set_pitch_and_yaw_async(pitch_deg, yaw_deg, callback);
 }
 
 Gimbal::Result GimbalImpl::set_pitch_rate_and_yaw_rate(float pitch_rate_deg_s, float yaw_rate_deg_s)
 {
-    wait_for_protocol();
+    if (_gimbal_protocol == nullptr) {
+        LogErr() << "Gimbal plugin not prepared! Call `prepare()` first!";
+        return Gimbal::Result::Error;
+    }
 
     return _gimbal_protocol->set_pitch_rate_and_yaw_rate(pitch_rate_deg_s, yaw_rate_deg_s);
 }
 
 void GimbalImpl::set_pitch_rate_and_yaw_rate_async(
     float pitch_rate_deg_s, float yaw_rate_deg_s, Gimbal::ResultCallback callback)
 {
-    wait_for_protocol_async([=]() {
-        _gimbal_protocol->set_pitch_rate_and_yaw_rate_async(
-            pitch_rate_deg_s, yaw_rate_deg_s, callback);
-    });
+    if (_gimbal_protocol == nullptr) {
+        _parent->call_user_callback([callback]() {
+            LogErr() << "Gimbal plugin not prepared! Call `prepare()` first!";
+            callback(Gimbal::Result::Error);
+        });
```

Comment:
And here. But place the `LogErr()` outside of the callback.

And the same below.